### DISCUSSION TITLE
Plugin Support for Runtime Dependencies

### DISCUSSION THREAD

**Discussion by maksim-romanov** [jpudysz/react-native-unistyles#265] [Feature Requests]
**Description:**
First of all, thank you for the great work on the unistyles library! It has been incredibly helpful for styling my React Native application ❤️ 🙏

**Feature Request:**
I would like to suggest adding a plugin system that allows developers to integrate various runtime dependencies into unistyles. The idea is to support runtime additions such as keyboard handlers, internet connection listeners, and other listeners or utilities that are commonly needed in React Native apps.

**Use Case:**
In many React Native projects, developers often need to work with platform-specific listeners (e.g., keyboard events, network status) or utility functions. Currently, integrating these into unistyles isn’t straightforward, and adding this feature would greatly enhance the flexibility and extendability of the library.

**Proposed Solution:**
Introduce a plugin system that would allow users to register and manage these runtime dependencies. Developers could then easily add plugins to handle specific scenarios (like monitoring keyboard state, network connectivity, etc.) directly within their unistyles setup.

I believe this would be a valuable addition and would make unistyles even more powerful for React Native developers.

Thank you for considering this feature!

**Comment by jpudysz** [comment]
Hey @maksim-romanov and thank you for the kind words.

> I would like to suggest adding a plugin system that allows developers to integrate various runtime dependencies into unistyles. The idea is to support runtime additions such as keyboard handlers, internet connection listeners, and other listeners or utilities that are commonly needed in React Native apps.

Who gonna write this handlers? How would you use eg. network listener with unistyles 🤔?

> Introduce a plugin system that would allow users to register and manage these runtime dependencies. Developers could then easily add plugins to handle specific scenarios (like monitoring keyboard state, network connectivity, etc.) directly within their unistyles setup.

Natively or from JS?

Remember that Unistyles is all about UI!
